### PR TITLE
feat(kserve): update domain template to include a namespace level (opendatahub-io/kserve#96)

### PR DESCRIPTION
**Pull request by aslakknutsen**
## Description
Adds a Namespace level to the KServe generated model routes.

## How Has This Been Tested?
Manual.

* Verified config change updates the correct CM, config-network in knative-serving
* Verified that the ISCV use the new domain template when generating Status.URL

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: pkg/feature/templates/serverless/serving-install/knative-serving.tmpl [hunk 1]
```diff
@@ -21,3 +21,5 @@ spec:
   config:
     istio:
       local-gateway.knative-serving.knative-local-gateway: "knative-local-gateway.{{ .ControlPlane.Namespace }}.svc.cluster.local"
+    network:
+      domainTemplate: "{{.Name}}.{{.Namespace}}.{{.Domain}}"
```
Comment (israel-hdez) [inline pkg/feature/templates/serverless/serving-install/knative-serving.tmpl:25]: I defer to @retocode, @skonto and/or @rhuss for a review.
They are the right people to tell if this change is still under support.

Comment (ReToCode) [reply pkg/feature/templates/serverless/serving-install/knative-serving.tmpl:25]: Hm why would you want to make this configurable in the first place? 
In theory this is (or could be) supported, but as we have an additional layer of routing in OpenShift which also needs to match the configuration we have in Knative. Depending on what people put there, OCP routing could stop working, so I'd vote to not change it. 

Btw, the flag would be `domain-template` using the newer syntax.

Comment (skonto) [reply pkg/feature/templates/serverless/serving-install/knative-serving.tmpl:25]: AFaik we changed the default downstream in order to align with OCP [conventions](https://issues.redhat.com/browse/SRVKS-56).

**Comment by zdtsw** [comment]
do we still need this draft?

**Comment by aslakknutsen** [comment]
@zdtsw Sorry, no. :)

**Pull request closed without merging**